endpoints/names/index.js

```diff
@@ -74,43 +72,49 @@ app.get('/names', (req, res) => {
 })
 
 /* Get all legal names for males */
-app.get('/names/males/:filter?', (req, res) => {
+app.get('/names/males/:search?', (req, res) => {
   const url = 'https://www.island.is/mannanofn/leit-ad-nafni/?Stafrof=&Drengir=on&Samthykkt=yes'
   return handleRequest(url, req, res)
 })
 
 /* Get all legal names for females */
-app.get('/names/females/:filter?', (req, res) => {
+app.get('/names/females/:search?', (req, res) => {
   const url = 'https://www.island.is/mannanofn/leit-ad-nafni/?Stafrof=&Stulkur=on&Samthykkt=yes'
   return handleRequest(url, req, res)
 })
 
 /* Get all legal middle names */
-app.get('/names/middlenames/:filter?', (req, res) => {
+app.get('/names/middlenames/:search?', (req, res) => {
   const url = 'https://www.island.is/mannanofn/leit-ad-nafni/?Stafrof=&Millinofn=on&Samthykkt=yes'
   return handleRequest(url, req, res)
 })
 
 /* Get all rejected names for males */
-app.get('/names/rejected/males/:filter?', (req, res) => {
+app.get('/names/rejected/males/:search?', (req, res) => {
   const url = 'https://www.island.is/mannanofn/leit-ad-nafni/?Stafrof=&Drengir=on&Samthykkt=no'
   return handleRequest(url, req, res)
 })
 
 /* Get all rejected names for females */
-app.get('/names/rejected/females/:filter?', (req, res) => {
+app.get('/names/rejected/females/:search?', (req, res) => {
   const url = 'https://www.island.is/mannanofn/leit-ad-nafni/?Stafrof=&Stulkur=on&Samthykkt=no'
   return handleRequest(url, req, res)
 })
 
 /* Get all rejected middle names */
-app.get('/names/rejected/middlenames/:filter?', (req, res) => {
+app.get('/names/rejected/middlenames/:search?', (req, res) => {
   const url = 'https://www.island.is/mannanofn/leit-ad-nafni/?Stafrof=&Millinofn=on&Samthykkt=no'
   return handleRequest(url, req, res)
 })
 
 /* Get all rejected names */
-app.get('/names/rejected/:filter?', (req, res) => {
+app.get('/names/rejected/:search?', (req, res) => {
   const url = 'https://www.island.is/mannanofn/leit-ad-nafni/?Stafrof=&Stulkur=on&Drengir=on&Millinofn=on&Samthykkt=no'
   return handleRequest(url, req, res)
 })
+
+/* Get all names with a given search */
+app.get('/names/:search', (req, res) => {
```

Comment:
Could you add this to the `/names` response above as well?

Reply:
Totally

Reply:
Small misunderstanding, but I actually like this much better and makes much more sense! We'd need some more eyes on this now though.

@koddsson @kristjanmik: See 8557e06.

Reply:
Small misunderstanding, but I actually like this much better and makes much more sense! We'd need some more eyes on this now though.

@koddsson @kristjanmik: See 8557e06.

Reply:
At first I thought you were pointing out the position of the '/names/:search' endpoint. If so, it is because when you use '/names/rejected' endpoint then the word 'rejected' will fit to :search parameter if it will be one of the first endpoints.

Reply:
Oh man, I understand why this was confusing. Had to read this whole file again to understand what I (probably) meant.

I might have meant to make the `:search` optional, so that there can be a combined handler for `/names` and `/name/:search`.

```suggestion
app.get('/names/:search?', (req, res) => {
```

Reply:
That's right, I don't know why I wanted to have the root router first but making it optional simplify the code. But the key is having that route below other so when someone call `/names/middlenames` will not trigger the `/names/:search?` and `'middlenames'` as a search query.

I'll fix that.